librosa/core/notation.py

```diff
@@ -124,15 +124,30 @@
 
 # Pre-compiled regular expressions for note and key parsing
 KEY_RE = re.compile(
-    r"^(?P<tonic>[A-Ga-g])" r"(?P<accidental>([#♯b!♭𝄪𝄫♮]*))" r":(?P<scale>(maj|min)(or)?)$"
+    r"^(?P<tonic>[A-Ga-g])"
+        r"(?P<accidental>[#♯𝄪b!♭𝄫]*)"
+        r":((?P<scale>(maj|min)(or)?)|(?P<mode>(ion|dor|phr(yg)?|lyd|mix(olyd)?|aeol?|locr?)(ian)?))$"
```

Comment:
I think this is probably fine, but it does accept some bad strings:
- `E:phrian`
- `G:mixian`
- `A:aeoian`
- `B:locian`
etc.

This is totally my fault :sweat_smile: as I was hacking things quickly to get a prototype implemented.  However, I think it's worth taking a minute to clearly think about what the grammar for key signatures should be going forward, and what abbreviations we should accept.

Reply:
I noticed that too.
One thought to organize things a bit: perhaps we allow only the spellings with the first three letters or without the last three, e.g. "mixolyd" and "mix" are allowed, and no other intermediates like "mixo".
At the risk of making the regex less readable, we could separate out the worrisome cases like so:
```suggestion
        r":((?P<scale>(maj|min)(or)?)|(?P<mode>(((ion|dor|phryg|lyd|mixolyd|aeol|locr)(ian)?)|phr|mix|aeo|loc)))$"
```